frontend/src/pages/login/LoginPage.tsx

```diff
@@ -1,14 +1,212 @@
 import { useAuth } from '~contexts/AuthContext'
-import Button from '~components/Button'
+import { FORM_GUIDE, REPORT_VULNERABILITY } from '~constants/externalLinks'
+import { LANDING_ROUTE } from '~constants/routes'
+import Footer from '~components/Footer'
+import Link from '~components/Link'
+
+import { LoginForm, LoginFormInputs } from './components/LoginForm'
+import { OtpForm, OtpFormInputs } from './components/OtpForm'
+
+export type LoginOtpData = {
+  email: string
+}
+
+const BrandLogo = chakra(BrandLogoSvg)
+const LoginImage = chakra(LoginImageSvg)
+
+export const LoginPage: FC = () => {
+  const [email, setEmail] = useState<string>()
+  const { sendLoginOtp, verifyLoginOtp } = useAuth()
+
+  const currentYear = new Date().getFullYear()
+  const isDesktop = useBreakpointValue({ base: false, xs: false, lg: true })
+  const isTablet = useBreakpointValue({ base: false, xs: false, md: true })
+
+  const footerLinks = useMemo(
+    () => [
+      { label: 'Contact Us', href: '/contact-us' },
+      { label: 'Guide', href: FORM_GUIDE },
+      { label: 'Privacy', href: '/privacy' },
+      { label: 'Terms of Use', href: '/terms' },
+      {
+        label: 'Report Vulnerability',
+        href: REPORT_VULNERABILITY,
+      },
+    ],
+    [],
+  )
+
+  const handleSendOtp = async ({ email }: LoginFormInputs) => {
+    await sendLoginOtp(email)
+    return setEmail(email)
+  }
+
+  const handleVerifyOtp = ({ otp }: OtpFormInputs) => {
+    // Should not happen, since OtpForm component is only shown when there is
+    // already an email state set.
+    if (!email) {
+      throw new Error('Something went wrong')
+    }
+    return verifyLoginOtp({ otp, email })
+  }
+
+  const handleResendOtp = async () => {
+    // Should not happen, since OtpForm component is only shown when there is
+    // already an email state set.
+    if (!email) {
+      throw new Error('Something went wrong')
+    }
+    await sendLoginOtp(email)
+  }
 
-export const LoginPage = (): JSX.Element => {
-  const { login } = useAuth()
   return (
-    <Box>
-      <Heading>This is a mock login page</Heading>
-      <Button onClick={login}>Log in</Button>
-    </Box>
+    <Flex flexDir="column" minH="100vh">
+      <Box
+        flexGrow={1}
+        px={{ base: '1.5rem', md: '5.5rem', lg: 0 }}
+        bg={{
+          base: 'initial',
+          md: 'linear-gradient(180deg, var(--chakra-colors-primary-500) 20.625rem, white 0)',
+          lg: 'linear-gradient(90deg, var(--chakra-colors-primary-500) 42%, white 0)',
+        }}
+      >
+        <Grid
+          minH={{ base: 'initial', lg: '100vh' }}
+          maxW="90rem"
+          margin="auto"
+          templateAreas={{
+            base: `'login'`,
+            md: `'sidebar' 'login'`,
+            lg: `'sidebar login' 'copy links'`,
+          }}
+          templateRows={{ lg: '1fr auto' }}
+          templateColumns={{ lg: '5fr 7fr' }}
```

Comment:
do you think there might be a better way to manage all these styles instead of inlining them? the component is damn hard to read rn, there's so much CSS that it's hard to get a sense of what the general structure of the component is and how the component state interacts with the DOM. might it be worth extracting all this CSS into a separate `ts` file so we can just spread in the component?

Reply:
We'll lose autosuggestions if that's the case. However, I can do it styled components style and create new components that are more indicative of what they do

Reply:
Done in 99371c9, is this more or less what you were looking for?